### ISSUE TITLE
How to stop loading m3u8 files (HTML5 and Flash)

### ISSUE THREAD

**Issue by FDiskas** [flowplayer/flowplayer-hlsjs#11]
Hello, I need to stop requesting m3u8 files in some conditions, but not stop the currently playing stream.
In some cases my playlist (m3u8) file appends the END tag and life stream becomes as an record.
Here is a sandbox: http://jsbin.com/gahusicemu/edit?js,console,output

I tried with no success to pass in config:
`
loader : customLoader,
xhrSetup : XMLHttpRequestSetupCallback,
`
Any suggestions are welcome.

**Comment by FDiskas** [comment]
If video is not playing I suggest you to install a chrome browser extention: https://chrome.google.com/webstore/detail/allow-control-allow-origi/nlfbmbojpeacfghkpbjhddihlkkiljbi?utm_source=chrome-app-launcher-info-dialog

**Comment by FDiskas** [comment]
```
{
        autoplay: true,
        adaptiveRatio: true,
        debug: true,
        ratio: 9 / 16,
        tooltip: false,
        keyboard: false,
        fullscreen: false,
        embed: false,
        clip: {
            sources: [
                {
                    type: "application/x-mpegurl",
                    src: "http://94.75.205.67/vytis/le.m3u8"
                }
            ],
            title: "Javascript setup (autoplay true)"
        },
        hlsjs: {
            debug: true,
            loader: function () {
                this.load = function (url, responseType, onSuccess, onError, timeout, maxRetry, retryDelay) {
                    console.log('Custom loader:', url);
                };

                /* abort any loading in progress */
                this.abort = function () {};
                /* destroy loading context */
                this.destroy = function () {};
            }
        }
    });
```
If I pass this config, then no requests is made - this is what i need. But if there is flash version still requesting m3u8 files.

**Comment by blacktrash** [comment]
I know that there's a Chrome extension which sort of implements CORS on the client side, sort of. But addon scenarios are not really something to be 'supported'.
If you want the stream to be played with hlsjs you should set the CORS policy on the server: https://github.com/flowplayer/flowplayer-hlsjs#cors
If you don't want to use the hlsjs engine in a specific player, simply set `hlsjs: false`.
Finally: I don't really understand the scenario: why not stop, or, even better, with a splash setup, unload the player? It looks like you custom loader only destroys the hlsjs instance, and the player then fails over to the flash hls engine. And in other clients to generic HLS.

**Comment by blacktrash** [comment]
ok, I can see the refreshed m3u8 being downloaded when you call only stop.
This is not hlsjs specific, it also happens with flash hls at http://demos.flowplayer.org/basics/live.html , so this should be addressed in the core player first.
This is live stream specific, and I'm not 100% sure whether this would be the correct behaviour on stop, and what the consequences are, but we can certainly look into it.
Pragmatically, as I said before you can use a splash setup and unload the player in such a situation.

Aside: of course we would be grateful if you enabled CORS for that stream, then we could show it in our demo ;-)

**Comment by FDiskas** [comment]
Please take a look at this screen video. Here is my problem. I got a playlist (m3u8) file smaller than before requested and then video stops playing. But if I force to seek `flowplayer().seek(1)` it plays again.
https://youtu.be/YGyQYK_Jnwc
And doration is wrong - not updating. If i try `flowplayer().video.duration = 2000` then seek-bar it updates correctly but not the total time displayed in controls-bar

**Comment by blacktrash** [comment]
@FDiskas - the YouTube video is private, I can't access it.
But assuming that we are talking about the stream at http://jsbin.com/gahusicemu/edit?js,console,output - this is a live stream, and it plays fine if you set `live: true`. You configured `live: false`, probably to get to the DVR functionality. But the DVR functionality is not available (yet) in the core player (if you follow our forums you know that), nevermind in this plugin.
The HLS DVR functionality must first be implemented in core for generic HLS and Flash HLS, then I'll see what's required to make it work in the hlsjs plugin.

I've opened a feature request for live DVR here: https://github.com/flowplayer/flowplayer/issues/950

Closing.
